[Build] Warnings when building with -Wswitch enabled on ESP32

When building with compiler switch `-Wswitch` (recently added in an open PR on ESP32 builds) these warnings are shown:
ESPeasySerial library:
```
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp: In function 'bool pinsChanged(ESPEasySerialPort, int, int)':
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:33:10: warning: enumeration value 'not_set' not handled in switch [-Wswitch]
   switch (port) {
          ^
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:33:10: warning: enumeration value 'sc16is752' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:33:10: warning: enumeration value 'serial0_swap' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:33:10: warning: enumeration value 'software' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:33:10: warning: enumeration value 'MAX_SERIAL_TYPE' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp: In function 'void setPinsCache(ESPEasySerialPort, int, int)':
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:47:10: warning: enumeration value 'not_set' not handled in switch [-Wswitch]
   switch (port) {
          ^
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:47:10: warning: enumeration value 'sc16is752' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:47:10: warning: enumeration value 'serial0_swap' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:47:10: warning: enumeration value 'software' not handled in switch [-Wswitch]
.pio/libdeps/custom_ESP32_4M316k/ESPeasySerial/ESPEasySerial_ESP32.cpp:47:10: warning: enumeration value 'MAX_SERIAL_TYPE' not handled in switch [-Wswitch]
```

and: 
```
src/src/Helpers_tmp/__tmpfile.cpp: In function 'String getPartitionType(uint8_t, uint8_t)':
src/src/Helpers_tmp/__tmpfile.cpp:7638:7: warning: case value '153' not in enumerated type 'esp_partition_subtype_t' [-Wswitch]
       case 0x99: return F("EEPROM"); // Not defined in esp_partition_subtype_t
       ^~~~
```

--- a/lib/ESPEasySerial/ESPEasySerial_ESP32.cpp
+++ b/lib/ESPEasySerial/ESPEasySerial_ESP32.cpp
@@ -36,6 +36,9 @@ bool pinsChanged(ESPEasySerialPort port,
     #ifndef ESP32S2
     case  ESPEasySerialPort::serial2: return receivePin != receivePin2 || transmitPin != transmitPin2;
     #endif
+    default:  
+      // No other hardware serial ports
+      break;
   }
   return false;
 }
@@ -61,6 +64,9 @@ void setPinsCache(ESPEasySerialPort port,
       break;
 
     #endif
+    default:  
+      // No other hardware serial ports
+      break;
   }
 }
 

--- a/lib/ESPEasySerial/ESPEasySerial_common.cpp
+++ b/lib/ESPEasySerial/ESPEasySerial_common.cpp
@@ -1,0 +1,29 @@
+#include <ESPeasySerial.h>
+
+
+String ESPeasySerial::getLogString() const {
+  String log;
+
+  log.reserve(48);
+  log = F("ESPEasy serial: ");
+
+  if (isI2Cserial()) {
+    log += F("I2C: addr:");
+    log += String(_receivePin);
+    log += F(" ch:");
+    log += _transmitPin == 0 ? 'A' : 'B';
+  } else {
+    if (isSWserial()) {
+      log += F("SW");
+    } else {
+      log += F("HW");
+    }
+    log += F(": rx:");
+    log += String(_receivePin);
+    log += F(" tx:");
+    log += String(_transmitPin);
+  }
+  log += F(" baud:");
+  log += String(_baud);
+  return log;
+}

--- a/lib/ESPEasySerial/ESPeasySerial_ESP8266.cpp
+++ b/lib/ESPEasySerial/ESPeasySerial_ESP8266.cpp
@@ -3,35 +3,7 @@
 
 #define DETECT_BAUDATE_TIMEOUT     250
 
-#include <Arduino.h>
 
-
-String ESPeasySerial::getLogString() const {
-  String log;
-
-  log.reserve(48);
-  log = F("ESPEasy serial: ");
-
-  if (isI2Cserial()) {
-    log += F("I2C: addr:");
-    log += String(_receivePin);
-    log += F(" ch:");
-    log += _transmitPin == 0 ? 'A' : 'B';
-  } else {
-    if (isSWserial()) {
-      log += F("SW");
-    } else {
-      log += F("HW");
-    }
-    log += F(": rx:");
-    log += String(_receivePin);
-    log += F(" tx:");
-    log += String(_transmitPin);
-  }
-  log += F(" baud:");
-  log += String(_baud);
-  return log;
-}
 
 // ****************************************
 // ESP8266 implementation wrapper

--- a/lib/ESPEasySerial/library.json
+++ b/lib/ESPEasySerial/library.json
@@ -1,6 +1,6 @@
 {
     "name": "ESPeasySerial",
-    "version": "2.0.8",
+    "version": "2.0.9",
     "keywords": [
       "serial", "io", "softwareserial", "hardwareserial"
     ],

--- a/lib/ESPEasySerial/library.json
+++ b/lib/ESPEasySerial/library.json
@@ -1,6 +1,6 @@
 {
     "name": "ESPeasySerial",
-    "version": "2.0.9",
+    "version": "2.0.10",
     "keywords": [
       "serial", "io", "softwareserial", "hardwareserial"
     ],

--- a/lib/ESPEasySerial/library.properties
+++ b/lib/ESPEasySerial/library.properties
@@ -1,5 +1,5 @@
 name=ESPeasySerial
-version=2.0.9
+version=2.0.10
 author=Gijs Noorlander
 maintainer=Gijs Noorlander <gijs.noorlander@gmail.com>
 sentence=Wrapper for SoftwareSerial, SC16IS752 I2C to UART bridge and HardwareSerial for ESP8266 and ESP32/ESP32S2.

--- a/lib/ESPEasySerial/library.properties
+++ b/lib/ESPEasySerial/library.properties
@@ -1,5 +1,5 @@
 name=ESPeasySerial
-version=2.0.8
+version=2.0.9
 author=Gijs Noorlander
 maintainer=Gijs Noorlander <gijs.noorlander@gmail.com>
 sentence=Wrapper for SoftwareSerial, SC16IS752 I2C to UART bridge and HardwareSerial for ESP8266 and ESP32/ESP32S2.

--- a/platformio_esp32_envs.ini
+++ b/platformio_esp32_envs.ini
@@ -34,7 +34,7 @@ lib_deps                  =
                             adafruit/Adafruit BusIO @ ^1.10.0
                             adafruit/Adafruit GFX Library@^1.11.1
                             adafruit/Adafruit ILI9341@^1.5.10
-                            td-er/ESPeasySerial @ 2.0.8
+                            td-er/ESPeasySerial @ 2.0.9
                             td-er/SparkFun MAX1704x Fuel Gauge Arduino Library @ ^1.0.1
 extra_scripts             = post:tools/pio/post_esp32.py
                             ${extra_scripts_default.extra_scripts}
@@ -44,6 +44,7 @@ build_flags               = ${core_esp32_IDF4_4__2_0_3.build_flags}
                             -DCONFIG_FREERTOS_ASSERT_DISABLE
                             -DCONFIG_LWIP_ESP_GRATUITOUS_ARP
                             -fno-strict-aliasing
+                            -Wswitch
                             -DCORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_NONE
 monitor_filters           = esp32_exception_decoder
 

--- a/platformio_esp32_envs.ini
+++ b/platformio_esp32_envs.ini
@@ -34,7 +34,7 @@ lib_deps                  =
                             adafruit/Adafruit BusIO @ ^1.10.0
                             adafruit/Adafruit GFX Library@^1.11.1
                             adafruit/Adafruit ILI9341@^1.5.10
-                            td-er/ESPeasySerial @ 2.0.9
+                            td-er/ESPeasySerial @ 2.0.10
                             td-er/SparkFun MAX1704x Fuel Gauge Arduino Library @ ^1.0.1
 extra_scripts             = post:tools/pio/post_esp32.py
                             ${extra_scripts_default.extra_scripts}

--- a/platformio_esp82xx_base.ini
+++ b/platformio_esp82xx_base.ini
@@ -68,7 +68,7 @@ lib_deps                  =
                             adafruit/Adafruit GFX Library@^1.11.1
                             adafruit/Adafruit ILI9341@^1.5.10
                             bblanchon/ArduinoJson @ ^6.17.2
-                            td-er/ESPeasySerial @ 2.0.9
+                            td-er/ESPeasySerial @ 2.0.10
                             td-er/RABurton ESP8266 Mutex @ ^1.0.2
                             td-er/SparkFun MAX1704x Fuel Gauge Arduino Library @ ^1.0.1
 lib_ignore                = ${esp82xx_defaults.lib_ignore}

--- a/platformio_esp82xx_base.ini
+++ b/platformio_esp82xx_base.ini
@@ -68,7 +68,7 @@ lib_deps                  =
                             adafruit/Adafruit GFX Library@^1.11.1
                             adafruit/Adafruit ILI9341@^1.5.10
                             bblanchon/ArduinoJson @ ^6.17.2
-                            td-er/ESPeasySerial @ 2.0.8
+                            td-er/ESPeasySerial @ 2.0.9
                             td-er/RABurton ESP8266 Mutex @ ^1.0.2
                             td-er/SparkFun MAX1704x Fuel Gauge Arduino Library @ ^1.0.1
 lib_ignore                = ${esp82xx_defaults.lib_ignore}

--- a/src/_C018.cpp
+++ b/src/_C018.cpp
@@ -584,6 +584,8 @@ bool CPlugin_018(CPlugin::Function function, struct EventStruct *event, String& 
       uint8_t stackVersion;
       uint8_t adr;
 
+      ESPEasySerialPort port = ESPEasySerialPort::not_set;
+
       {
         // Keep this object in a small scope so we can destruct it as soon as possible again.
         std::shared_ptr<C018_ConfigStruct> customConfig;
@@ -610,6 +612,7 @@ bool CPlugin_018(CPlugin::Function function, struct EventStruct *event, String& 
         joinmethod    = customConfig->joinmethod;
         stackVersion  = customConfig->stackVersion;
         adr           = customConfig->adr;
+        port = static_cast<ESPEasySerialPort>(customConfig->serialPort);
 
         {
           addFormTextBox(F("Device EUI"), F("deveui"), customConfig->DeviceEUI, C018_DEVICE_EUI_LEN - 1);
@@ -664,8 +667,7 @@ bool CPlugin_018(CPlugin::Function function, struct EventStruct *event, String& 
 
       addTableSeparator(F("Serial Port Configuration"), 2, 3);
 
-      // FIXME TD-er: Add port selector
-      serialHelper_webformLoad(ESPEasySerialPort::not_set, rxpin, txpin, true);
+      serialHelper_webformLoad(port, rxpin, txpin, true);
 
       // Show serial port selection
       addFormPinSelect(PinSelectPurpose::Generic_input, formatGpioName_RX(false),                   F("taskdevicepin1"), rxpin);

--- a/src/src/Helpers/ESPEasy_Storage.cpp
+++ b/src/src/Helpers/ESPEasy_Storage.cpp
@@ -1666,6 +1666,10 @@ String getPartitionType(uint8_t pType, uint8_t pSubType) {
     }
   }
 
+  if (partitionSubType == 0x99) {
+    return F("EEPROM"); // Not defined in esp_partition_subtype_t
+  }
+
   if (partitionType == ESP_PARTITION_TYPE_DATA) {
     switch (partitionSubType) {
       case ESP_PARTITION_SUBTYPE_DATA_OTA:      return F("OTA selection");
@@ -1680,7 +1684,6 @@ String getPartitionType(uint8_t pType, uint8_t pSubType) {
         #else
         return F("SPIFFS");
         #endif
-      case 0x99: return F("EEPROM"); // Not defined in esp_partition_subtype_t
       default: break;
     }
   }

